getgauge/impl_loader.py

```diff
@@ -97,11 +97,14 @@ def _import_file(base_dir, file_path):
 
 # Inject instace in each class method (hook/step)
 def update_step_resgistry_with_class(instance, file_path):
-    for info in registry.get_all_methods_in(file_path):
+    # Resolve the absolute path from relative path
+    file_path = os.path.abspath(file_path) if '..' in file_path else file_path
```

Comment:
Thanks for the test! Can you explain why you added the if check for ".." because `os.path.abspath(file_path)` resolves to an absolute path. So technically you can resolve everything to it's absolute path.

Reply:
Hey @zabil,

I added the '..' check to see if the `file_path` is a relative one and only then resolve it to the absolute path. This way we save some time it takes for the `os.path.abspath(file_path)`method to actually resolve the absolute path. I tried this without the if condition in my project which has lots of impl files to load and it took a little more time to load all methods and start the execution compared to when I had the condition. That's the only reason, please let me know if I should remove it, won't be an issue.

Thanks!😄

Reply:
Ah ok. Fair point.